pkg/webhook/admission/validator_test.go

```diff
@@ -0,0 +1,280 @@
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+var _ = Describe("validatingHandler", func() {
+	Describe("Handle", func() {
+		It("should return 200 in response when create succeeds", func() {
+
+			handler := createSucceedingValidatingHandler()
+
+			response := handler.Handle(context.TODO(), Request{
+				AdmissionRequest: v1beta1.AdmissionRequest{
+					Operation: v1beta1.Create,
+					Object: runtime.RawExtension{
+						Raw:    []byte("{}"),
+						Object: handler.validator,
+					},
+				},
+			})
+			Expect(response.Allowed).Should(BeTrue())
+			Expect(response.Result.Code).Should(Equal(int32(http.StatusOK)))
+		})
+
+		It("should return 400 in response when create fails on decode", func() {
+			//TODO
+		})
+
+		It("should return response built with the Status object when ValidateCreate returns APIStatus error", func() {
+
+			handler, expectedError := createValidatingHandlerWhichReturnsStatusError()
+
+			response := handler.Handle(context.TODO(), Request{
+				AdmissionRequest: v1beta1.AdmissionRequest{
+					Operation: v1beta1.Create,
+					Object: runtime.RawExtension{
+						Raw:    []byte("{}"),
+						Object: handler.validator,
+					},
+				},
+			})
+			Expect(response.Allowed).Should(BeFalse())
+
+			apiStatus, ok := expectedError.(apierrs.APIStatus)
+			Expect(ok).Should(BeTrue())
+			Expect(response.Result.Code).Should(Equal(apiStatus.Status().Code))
+			Expect(*response.Result).Should(Equal(apiStatus.Status()))
+
+		})
+
+		It("should return 403 response when ValidateCreate returns non-APIStatus error", func() {
+
+			handler, expectedError := createValidatingHandlerWhichReturnsRegularError()
+
+			response := handler.Handle(context.TODO(), Request{
+				AdmissionRequest: v1beta1.AdmissionRequest{
+					Operation: v1beta1.Create,
+					Object: runtime.RawExtension{
+						Raw:    []byte("{}"),
+						Object: handler.validator,
+					},
+				},
+			})
+			Expect(response.Allowed).Should(BeFalse())
+			Expect(response.Result.Code).Should(Equal(int32(http.StatusForbidden)))
+			Expect(string(response.Result.Reason)).Should(Equal(expectedError.Error()))
+
+		})
+
+		It("should return 200 in response when update succeeds", func() {
+
+			handler := createSucceedingValidatingHandler()
+
+			response := handler.Handle(context.TODO(), Request{
+				AdmissionRequest: v1beta1.AdmissionRequest{
+					Operation: v1beta1.Update,
+					Object: runtime.RawExtension{
+						Raw:    []byte("{}"),
+						Object: handler.validator,
+					},
+					OldObject: runtime.RawExtension{
+						Raw:    []byte("{}"),
+						Object: handler.validator,
+					},
+				},
+			})
+			Expect(response.Allowed).Should(BeTrue())
+			Expect(response.Result.Code).Should(Equal(int32(http.StatusOK)))
+		})
+
+		It("should return 400 in response when update fails on decoding new object", func() {
```

Comment:
these should be marked `PIt` to mark them as "pending" (not-yet-written" tests)

Also, don't leave TODOs without a name (yours) or bug number without context attached.

Reply:
Fixed